tests/test_token_header_auth.py

```diff
@@ -31,8 +31,18 @@ def header_test_endpoint(header_str: str = Depends(
     return header_str
 
 
-@router.get("/header/token/dict")
-def header_test_endpoint_decode(payload_dict: Dict[str, Any] = Depends(
+@router.get("/header/token/optional/str")
+def optional_header_test_endpoint(header_str: str = Depends(
+    auth_util.get_token_from_optional_header)):
+    """
+    Tests the same endpoint as above but with the optional instead of
+    required return function.
+    """
+    return header_str
```

Comment:
Hadn't considered the need for an optional header, will definitely come in handy. Great PR